src/main/scala/com/wavesplatform/mining/Miner.scala

```diff
@@ -27,15 +27,26 @@ import scorex.wallet.Wallet
 import scala.collection.mutable.{Map => MMap}
 import scala.concurrent.Await
 import scala.concurrent.duration._
+import MinerDebugInfo.State
 
 trait Miner {
   def scheduleMining(): Unit
 }
 
 trait MinerDebugInfo {
+  def state: State.Value
   def collectNextBlockGenerationTimes: List[(Address, Long)]
 }
 
+object MinerDebugInfo {
+  object State extends Enumeration {
+    val Disabled = Value("disabled")
+    val Error = Value("error")
+    val Mining_Blocks = Value("mining blocks")
```

Comment:
Please mind  `CamelCase` and make it sealed trait and case objects, no need for value(..). No need for enumeration if you don't enumerate anything.